common/djangoapps/third_party_auth/models.py

```diff
@@ -827,6 +848,77 @@ def get_setting(self, name):
             return other_settings[name]
         raise KeyError
 
+    def get_current_saml_configuration(self):
+        """
+        Get the current active SAMLConfiguration for this provider.
+        Falls back to the default configuration if none is set.
+
+        Provides observability through custom attributes regardless of toggle state.
+        """
+        # Always provide observability, regardless of toggle state
+        signal_handlers_enabled = ENABLE_SAML_CONFIG_SIGNAL_HANDLERS.is_enabled()
+        signal_handlers_status = 'enabled' if signal_handlers_enabled else 'disabled'
+        set_custom_attribute('saml_config.signal_handlers', signal_handlers_status)
```

Comment:
Simplifying code and ability for custom attributes to match code.
```suggestion
        signal_handlers_enabled = ENABLE_SAML_CONFIG_SIGNAL_HANDLERS.is_enabled()
        set_custom_attribute('saml_config.signal_handlers_enabled', signal_handlers_enabled)
```

Reply:
Implemeted